compose/compiler/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/ComposeIrGenerationExtension.kt

```diff
@@ -121,6 +126,18 @@ class ComposeIrGenerationExtension(
 
         val mangler = when {
             pluginContext.platform.isJs() -> JsManglerIr
+            // Use Reflection to get KonanManglerIr from konan
+            // TODO: Use `import` to get org.jetbrains.kotlin.backend.konan.* modules
+            pluginContext.platform.isNative() -> {
```

Comment:
I guess we need to access KonanManglerIr and KonanGlobalDeclarationTable for k/n only with decoys enabled. JB's builds of Compose for K/N don't use decoys now. So maybe adding one more condition `decoysEnabled` would be helpful. 

Another thought: ideally, it would be accessible without reflection.  Maybe it's worth submitting a ticket at https://youtrack.jetbrains.com/ describing the use case and linking it here in a comment, so we don't forget to update/improve this part later.

Reply:
Maybe, since, both `mangler` and `idSignatureBuilder` needed only when decoys enabled, we can have two functions (somewhere in decoys sub-package) to create mangler and idSignatureBuilder instances. This would make this class a bit lighter, I guess.